sparse_ho/ho.py

```diff
@@ -13,34 +13,41 @@ def grad_search(
 
     Parameters
     ----------
-    algo: TODO
-    criterion: TODO
-    model: TODO
+    algo: Algo object
+        algorithm used to compute hypergradient.
+    criterion: Criterion object
+        criterion to optimize during hyperparameter optimization
+        (outer optimization problem).
+    model: Model Object
+        model on which hyperparameter has to be selected
+        (inner optimization problem).
+    X: {ndarray, sparse matrix} of (n_samples, n_features)
+        Data.
+    y: {ndarray, sparse matrix} of shape (n_samples,)
+        Target.
     log_alpha0: float
-        log of the regularization coefficient alpha
-    tol : float
-        tolerance for the inner optimization solver
+        log of the regularization coefficient alpha.
     monitor: Monitor object
-        used to store the value of the cross-validation function
+        used to store the value of the cross-validation function.
     n_outer: int
-        number of maximum iteration in the outer loop (for the line search)
+        Defaults to 100.
+        number of maximum iteration in the outer loop (for the line search).
+    verbose: Boolean
```

Comment:
```suggestion
    verbose: bool
```
Boolean is not a Python type